pkg/workloadmeta/collectors/internal/kubeapiserver/deployments.go

```diff
@@ -0,0 +1,85 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package kubeapiserver
+
+import (
+	"context"
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/workloadmeta"
+)
+
+const deploymentStoreName = "deployments-store"
+
+func init() {
+	resourceSpecificGenerator[deploymentStoreName] = newDeploymentStore
+}
+
+func newDeploymentStore(ctx context.Context, cfg config.Config, wlm workloadmeta.Store, client kubernetes.Interface) (*cache.Reflector, *reflectorStore, error) {
+	if !cfg.GetBool("language_detection.enabled") {
+		return nil, nil, fmt.Errorf("language detection is enabled") // we might remove this if we want to use deployments info in unified service tagging
+	}
+	deploymentListerWatcher := &cache.ListWatch{
+		ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+			return client.AppsV1().Deployments(metav1.NamespaceAll).List(ctx, options)
+		},
+		WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+			return client.AppsV1().Deployments(metav1.NamespaceAll).Watch(ctx, options)
+		},
+	}
+
+	deploymentStore := newDeploymentReflectorStore(wlm)
+	deploymentReflector := cache.NewNamedReflector(
+		componentName,
+		deploymentListerWatcher,
+		&appsv1.Deployment{},
+		deploymentStore,
+		noResync,
+	)
+	return deploymentReflector, deploymentStore, nil
+}
+
+func newDeploymentReflectorStore(wlmetaStore workloadmeta.Store) *reflectorStore {
+	store := &reflectorStore{
+		wlmetaStore: wlmetaStore,
+		seen:        make(map[string]workloadmeta.EntityID),
+		parser:      newdeploymentParser(),
+	}
+
+	return store
+}
+
+type deploymentParser struct{}
+
+func newdeploymentParser() objectParser {
+	return deploymentParser{}
+}
+
+func (p deploymentParser) Parse(obj interface{}) workloadmeta.Entity {
+	deployment := obj.(*appsv1.Deployment)
+
+	return &workloadmeta.KubernetesDeployment{
+		EntityID: workloadmeta.EntityID{
+			Kind: workloadmeta.KindKubernetesDeployment,
+			ID:   deployment.Name, // not sure if we should use the UID or the name here
+		},
+		EntityMeta: workloadmeta.EntityMeta{
+			Name:      deployment.Name,
+			Namespace: deployment.Namespace,
+			Labels:    deployment.Labels,
+		},
```

Comment:
I don't the the containers' process language information in this entity. 

I think instead of storing all the labels we can already pre-aggregate  the information to be usable

Reply:
It makes sense, we can define which fields we need in the struct

Reply:
In Unified Service Tagging we need all the labels. Also I don't use the annotations but shouldn't we store language info as annotation ?

Reply:
After digging into UST code, we do not need every label.